src/gssw.h

```diff
@@ -22,6 +22,47 @@
 struct gssw_profile;
 typedef struct gssw_profile gssw_profile;
 
+// This file makes extensive use of SSE intrinsics, which operate on 128-bit
+// values. These values are stored in the 128-bit special registers XMM0 through
+// XMM7, as assigned by the compiler. In C, these values have type "__m128i",
+// with the "i" meaning integer (because we do integer math on them). We have
+// code to use these in alignments, either to store 16 single-byte values or 8
+// double-byte values, depending on how large we're allowing the scores to get.
+
+// We use j for indexes in the read (or "query"), and i for indexes in the
+// reference (or "database"). Conceptually, the database runs rightwards along
+// the top of the DP matrix, and the read runs downwards along the left.
+
+// Note that this usage of i and j is BACKWARDS from the paper!
```

Comment:
I assume you mean the SSW paper?

Also, we should write this up. @adamnovak I think this work rationalizes gssw and completes the project; pending some testing via vg.